pypeit/scripts/collate_1d.py

```diff
@@ -0,0 +1,823 @@
+        else:
+            self.coord = spec1d_obj['SPAT_PIXPOS']
+
+        self.coaddfile = self.build_coadd_file_name()
+
+    def build_coadd_file_name(self):
+        """Build the output file name for coadding.
+        The filename convention is J<hmsdms+dms>_<instrument name>_<YYYYMMDD>.fits
+        when matching by RA/DEC and SPAT_<spatial position>_<instrument name>_<YYYYMMDD>.fits
+        when matching by pixel position..
+
+        Currently instrument_name is taken from spectrograph.camera
+
+        Return: str  The name of the coadd output file.
+        """
+        time_portion = Time(self.spec1d_header_list[0]['MJD'], format="mjd").strftime('%Y%m%d')
+        if self.match_type == 'ra/dec':
+            coord_portion = 'J' + self.coord.to_string('hmsdms', sep='', precision=2).replace(' ', '')
+        else:
+            coord_portion = self.spec_obj_list[0]['NAME'].split('_')[0]
+        instrument_name = self._spectrograph.camera
+
+        return f'{coord_portion}_{instrument_name}_{time_portion}.fits'
+
+    def _config_key_match(self, header):
+        """
+        Check to see if the configuration keys from a spec1d file match the
+        ones for this SourceObject.
+
+        Args:
+        header (:obj:`astropy.io.fits.Header`):
+            Header from a spec1d file.
+
+        Returns (bool): True if the configuration keys match, 
+            false if they do not.
+        """
+        # Make sure the spectrograph matches
+        if 'PYP_SPEC' not in header or header['PYP_SPEC'] != self._spectrograph.name:
+            return False
+
+        first_header = self.spec1d_header_list[0]
+
+        for key in self._spectrograph.configuration_keys():
+            # Ignore "decker" because it's valid to coadd spectra with different slit masks
+            if key != "decker":
+                if key not in first_header and key not in header:
+                    # Both are missing the key, this is ok
+                    continue
+                elif key not in first_header or key not in header:
+                    # One has a value and the other doesn't so they don't match
+                    return False
+
+                if first_header[key] != header[key]:
+                    return False
+
+        # No mismatches were found
+        return True
+
+    def match(self, spec_obj, spec1d_header, tolerance):
+        """Determine if a SpecObj matches this group within the given tolerance.
+        This will also compare the configuration keys to make sure the SpecObj
+        is compatible with the ones in this SourceObject.
+
+        Args:
+        spec_obj (:obj:`pypeit.specobj.SpecObj`): 
+            The SpecObj to compare with this SourceObject.
+
+        spec1d_header (:obj:`astropy.io.fits.Header`):
+            The header from the spec1d that dontains the SpecObj.
+            
+        tolerance (float OR :obj:`astropy.coordinates.Angle`):
```

Comment:
My preference for this argument is that it would not be an astropy.coordinates.Angle object but rather either be a float, or a astropy quantity with units of angle, i.e. u.arcsec or u.deg. But I guess that depends on whether users will be calling this function directly. My point is that there are slightly more lines of code required to instantiate an astropy angle object, as compared to just passing in the units of the matching tolerance.

Reply:
I changed it to take a floating point value and an optional "unit" argument that defaults to arcseconds.